docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,503 @@
+    kind: Mesh
+  from:
+    - targetRef:
+        kind: Mesh
+      conf:
+        action: DENY
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          kuma.io/zone: us-east
+      conf:
+        action: ALLOW
+---
+type: MeshTrafficPermission
+name: backend
+spec:
+  targetRef:
+    kind: MeshService
+    name: backend
+  from:
+    - targetRef:
+        kind: MeshSubset
+        tags:
+          env: dev
+      conf:
+        action: DENY
+```
+
+Concatenated "from" array
+```yaml
+from:
+  - targetRef:
+      kind: Mesh
+    conf:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        env: dev
+    conf:
+      action: DENY
+```
+
+2. Create a full rule-based view (with negations) 
+
+Regular [rule-based view](005-policy-matching.md#rule-based-view) doesn't work well for NetworkRBAC generation,
+because rule-based view relies on the order, but NetworkRBAC doesn't. 
+
+In this example, if a request matches "rule 2" this means it doesn't match "rule 1":
+```yaml
+rules:
+  - targetRef: # rule 1
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    conf:
+      action: DENY
+  - targetRef: # rule 2
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+    conf:
+      action: ALLOW
+```
+
+So "rule 2" implicitly matches requests from zone "us-east" that are **not** from env "dev".
+That's why it's important to have a full rule-based view with negations ("!" means "not"):
+
+```yaml
+rules:
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: dev
+    conf:
+      action: DENY
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: us-east
+        env: !dev
+    conf:
+      action: ALLOW
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: dev
+    conf:
+      action: DENY 
+  - targetRef:
+      kind: MeshSubset
+      tags:
+        kuma.io/zone: !us-east
+        env: !dev
+    conf:
+      action: DENY
+```
+
+See [how to generate a full rule-based view with negations](#how-to-generate-a-full-rule-based-view-with-negations)
+
+3. For each "rule" that has "ALLOW" action generate a separate NetworkRBAC policy
+
+```yaml
+action: ALLOW
+policies:
+  "mtp-1":
+    permissions:
+      - any: true
+    principals:
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+```
+
+4. If MeshTrafficPermission is targeting proxy then NetworkRBAC is placed as a NetworkFilter on the inbound listener.
+If MeshTrafficPermission is targeting MeshHTTPRoute then NetworkRBAC is placed as a NetworkFilter on the route.
```

Comment:
How will your "rule-based view with negations" work in this case? Do you just have 2 different ones one for route and one for listener?

Reply:
That's correct, there is per-listener and per-route RBAC. Added this to MADR